### PR TITLE
Rename from SPR to DLR

### PR DESCRIPTION
**Pull request by shinaoka**

### CODE REVIEW

File: src/sparse_ir/dlr.py [hunk 1]
```diff
@@ -10,16 +10,16 @@
 from . import svd
 
 
-class SparsePoleRepresentation(abstract.AbstractBasis):
+class DiscreteLehmannRepresentation(abstract.AbstractBasis):
     """
-    Sparse pole representation (SPR)
-    The poles are the extrema of V'_{L-1}(ω)
+    Discrete Lehmann representation (DLR)
+    The poles are choosen to be the extrema of V'_{L-1}(ω).
```
Comment (mwallerb) [inline src/sparse_ir/dlr.py:16]:
```suggestion
    Discrete Lehmann representation (DLR) with poles selected according to extrema of IR.
    
    This class implements a variant of the discrete Lehmann representation (`DLR`_).  Instead
    of a truncated singular value expansion of the analytic continuation kernel ``K`` like the IR,
    the discrete Lehmann representation is based on a "sketching" of ``K``.  The resulting basis
    is a linear combination of discrete set of poles on the real-frequency axis, continued to the
    imaginary-frequency axis:
    
         G(iv) == sum(a[i] / (iv - w[i]) for i in range(L))
    
    Warning
        The poles on the real-frequency axis selected for the DLR are based on a rank-revealing
        decomposition, which offers accuracy guarantees.  Here, we instead select the pole locations
        based on the zeros of the IR basis functions on the real axis, which is a heuristic.  We do not
        expect that difference to matter, but please don't blame the DLR authors if we were wrong :-)  
    
    [1]: https://doi.org/10.48550/arXiv.2110.06765
    
```

**Review by mwallerb — APPROVED** [review]
Agreed with this change.

Just as a comment, the original name was chosen to avoid confusion with DLR, which chooses the poles in a different way and may thus be superiour to this.  I think we should at least make this difference clear.  @jasonkaye is this text acceptable?

**Comment by shinaoka** [comment]
Approved by Jason.

**Comment by shinaoka** [comment]
Oh, I forgot to accept your suggestion. I will apply this manually...

**Pull request merged**